src/lib/y2storage/partition_tables/base.rb

```diff
@@ -114,6 +115,26 @@ def inspect
         "<PartitionTable #{self}[#{num_children}] #{parts}#{slots}>"
       end
 
+      # The partition id depends on the partition table type. For example,
+      # MSDOS partition tables use SWAP id for swap partitions, but DASD
+      # partiton tables always need LINUX id.
+      #
+      # With this method, each partition table can define the partition id
+      # it expects for each case.
+      #
+      # TODO: right now only :swap is considered.
+      #
+      # @param id_name [Symbol] a partition id name (e.g., :swap, :linux, :ntfs)
```

Comment:
I don't think this should have a direct correlation to partitions ids. If that's the case, why not simple accepting `<PartitionId>` as argument?

I believe the intention of this argument is more like a role. `:swap`, `lvm_pv`, `windows`...